src/types.ts

```diff
@@ -79,12 +83,44 @@ export interface PluginEvent {
     uuid?: string
 }
 
+// Event after being processed by PostHog ingestion pipeline.
+export interface ProcessedPluginEvent {
+    distinct_id: string
+    ip: string | null
+    team_id: number
+    event: string
+    properties: Properties
+    timestamp: string
+    /** Person properties update (override). */
+    $set?: Properties
+    /** Person properties update (if not set). */
+    $set_once?: Properties
+    /** The assigned UUIDT of the event (EE pipeline-only). */
+    uuid?: string
+}
+
 export interface PluginAttachment {
     content_type: string
     file_name: string
     contents: any
 }
 
+export interface Action {
+    id: number
+    team_id: number
+    name: string | null
+    description: string
+    created_at: string
+    created_by_id: number | null
+    deleted: boolean
+    post_to_slack: boolean
+    slack_message_format: string
+    is_calculating: boolean
+    updated_at: string
+    last_calculated_at: string
+    steps: any[]
+}
+
```

Comment:
let's maybe preemptively limit the contract here? let's leave the contract as the "meat" of an action to future-proof this a bit?

```suggestion
export interface Action {
    team_id: number
    name: string | null
    description: string
    steps: any[]
}
```

One could argue for `created_at` and `updated_at` as well but given nobody uses this yet could make sense to limit as much as possible